### DISCUSSION TITLE
[BUG] Multiple issues with GSC

### DISCUSSION THREAD

**Discussion by Ahmed-E-86** [HEIGE-PCloud/DoIt#339] [General]
### Describe the bug 
I have 2 issues with Google Search Console:

#### The first issue:
Google Search Console marked some pages as **not mobile friendly**, and I went to more info, and I found these warnings:

![2021-11-06 00 28 44 search google com b97cd6ebe462](https://user-images.githubusercontent.com/29783425/140587277-f238ea29-b12a-4e0d-835c-9b778ce2fa9b.png)

![2021-11-06 00 55 59 search google com ada85125da3c](https://user-images.githubusercontent.com/29783425/140587289-c63f7755-3a0a-4c08-b5ad-a0b392c5bd83.png)

#### The second issue:
- I migrated from an old domain to a new domain, and I wanted to keep some URLs as it used to be before, so I used url front matter to keep the old format.

**An example:**
```
---
url: /html/2021/1/name-of-the-file.html
--
```
But somehow **Google Search Console**, and **Yandex Webmaster** identify two pages for the same post, one with .html at the end, and one without .html at the end, and mark one of them as a duplicate.

I have no idea why I am having these issues, and how to fix them.

### Build Environment

- OS: Ubuntu 21.10
- Theme version/commit [0.2.13, d4e7dd2]
- Hugo version [0.89.1]

### Preview Environment

- OS: Ubuntu 21.10
- Browser: Vivaldi.

**Comment by HEIGE-PCloud** [comment]
For the first issue, I have unblocked all CSS and js file in the default robots.txt template. Now, the Google bot should be able to fetch them correctly.

https://github.com/HEIGE-PCloud/DoIt/commit/fa0fb98eb58aa7c2ad81680134a3b309a9167e48

**Reply by Ahmed-E-86** [reply]
Still the same :/

![Screenshot_20211106_170515](https://user-images.githubusercontent.com/29783425/140614320-9ed76474-d77b-4a12-a0ed-28d05dcd6a8f.png)

**Reply by HEIGE-PCloud** [reply]
Sorry, forget about the `Allow` field

https://github.com/HEIGE-PCloud/DoIt/commit/398a024845ec59571e0d421287eaf4a82a3c9b18

Please try again...

**Reply by Ahmed-E-86** [reply]
I did not receive any new errors. Thanks!

**Comment by HEIGE-PCloud** [comment]
For the second one, I am not so sure, but whether setting the url to

```
---
url: /html/2021/1/name-of-the-file
---
```

will solve it?

**Reply by Ahmed-E-86** [reply]
The issue that there are other URLs for the URLs end with .html, and I want to get rid of the URLs without .html, because it confuses search engines.

**Reply by HEIGE-PCloud** [reply]
Maybe https://gohugo.io/content-management/urls/#ugly-urls

**Reply by HEIGE-PCloud** [reply]
However, I believe url ends in .html will be automatically redirect to the url without .html

https://simpler-website.pages.dev/html/2020/12/proton-troubleshooting.html will be redirected to https://simpler-website.pages.dev/html/2020/12/proton-troubleshooting

So I think just keep it with no .html and everything will be fine (?

**Reply by Ahmed-E-86** [reply]
I checked one of the URLs end with .html, and I got this error:

![Screenshot_20211106_173953](https://user-images.githubusercontent.com/29783425/140615503-c8d6df00-35a9-4344-b937-018b86a5b8eb.png)

I guess URLs end with .html redirected to URLs without .html, and search engines disprove these links, but that did not happen before, these issues started to happen lately.

Also, I use Waline which records counters for visitors, and I noticed a month ago, that there are new records for the same old .html pages but without .html at the end, and I removed them, but I found them again with higher numbers that keeps growing.

**Reply by HEIGE-PCloud** [reply]
Waline uses `window.location.pathname` to create record, so if the redirect happens successfully to your visitors, then a url without .html will be recorded by Waline.

**Reply by Ahmed-E-86** [reply]
> Waline uses `window.location.pathname` to create record, so if the redirect happens successfully to your visitors, then a url without .html will be recorded by Waline.

But when I enter a URL ends with .html, Waline uses the record ends with .html not the new record.

Search engines recorded 2 URLs for the same posts, and if I remove the URL ended with .html, the redirected links will not work, so I have to keep it this way.

I feel that there is nothing to be done, because it will get worse whatever I do.

**Comment by Ahmed-E-86** [comment]
@HEIGE-PCloud Also, I found these SEO errors at Microsoft Bing Webmaster Tools:

![Screenshot_20211106_211558](https://user-images.githubusercontent.com/29783425/140621298-6f63adc5-5723-4cb7-8cd7-875391836d1f.png)
